### PR TITLE
[SPARK-14896][SQL] Deprecate HiveContext in python

### PR DESCRIPTION
**Pull request by andrewor14**
## What changes were proposed in this pull request?

See title.
## How was this patch tested?

PySpark tests.

### CODE REVIEW

File: python/pyspark/sql/column.py [hunk 1]
```diff
@@ -419,7 +419,7 @@ def over(self, window):
         >>> from pyspark.sql.functions import rank, min
         >>> # df.select(rank().over(window), min('age').over(window))
 
-        .. note:: Window functions is only supported with HiveContext in 1.4
+        .. note:: Window functions is only supported with Hive support in 1.4
```
Comment (rxin) [inline python/pyspark/sql/column.py]: remove this note entirely?

Comment (andrewor14) [inline python/pyspark/sql/column.py]: ha yeah IDK why this is in anything that's not branch-1.4

**Comment by rxin** [comment]
LGTM feel free to merge now since the last commit was trivial.

**Comment by andrewor14** [comment]
ok!

**Pull request closed without merging**